bgpd/bgp_routemap.c

```diff
@@ -403,16 +403,41 @@ route_match_script(void *rule, const struct prefix *prefix, void *object)
 		zlog_debug("Updating attribute based on script's values");
 
 		uint32_t locpref = 0;
+		uint32_t med = 0;
 
-		path->attr->med = newattr.med;
+		if (CHECK_FLAG(path->attr->flag,
+			       ATTR_FLAG_BIT(BGP_ATTR_MULTI_EXIT_DISC)))
+			med = path->attr->med;
+		if (med != newattr.med) {
+			path->attr->med = newattr.med;
+			SET_FLAG(path->attr->flag,
+				 ATTR_FLAG_BIT(BGP_ATTR_MULTI_EXIT_DISC));
+		}
+
+		path->attr->aspath = newattr.aspath;
```

Comment:
Hm, I think we need here aspath_unintern() and intern a new value? Same for other pointers: community, lcomm, extcomm (bgp_attr_intern()).